## Convert the TELL-Seq reads that were converted to 10x format or 10x reads into shotgun reads using the Scaff10x package
	conda create -n Singularity -c conda-forge singularity
	conda activate singularity

	sh /home/FM/thains/releases_v1.1_package/tellread-release/run_tellread_fq_sing.sh -i1 /home/FM/thains/TELLSeq_data/rawfastq/JB-TH-5S-PP_S5_R2_001.fastq.gz -r1 /home/FM/thains/TELLSeq_data/rawfastq/JB-TH-5S-PP_S5_R1_001.fastq.gz -r2 /home/FM/thains/TELLSeq_data/rawfastq/JB-TH-5S-PP_S5_R3_001.fastq.gz -o /home/FM/thains/TELLSeq_data/PyrPer1 -g NONE -s T504

# Convert TELL-Seq fastq files into 10x fastq files for Supernova
	/home/FM/thains/releases_v1.1_package/conversion_tool/ust10x -i1 /home/FM/thains/TELLSeq_data/bPyrPer1/Full/PyrrhuraPerlata_I1_T501.fastq.gz.corrected.err_barcode_removed.fastq.gz -r1 /home/FM/thains/TELLSeq_data/bPyrPer1/Full/PyrrhuraPerlata_R1_T501.fastq.gz.corrected.err_barcode_removed.fastq.gz -r2 /home/FM/thains/TELLSeq_data/bPyrPer1/Full/PyrrhuraPerlata_R2_T501.fastq.gz.corrected.err_barcode_removed.fastq.gz -wl /home/FM/thains/releases_v1.1_package/conversion_tool/4Mwith-alts-february-2016.txt
	gzip R1_sl.fastq.gz.4tenx.fastq
	mkdir /home/FM/thains/TELLSeq/convertedFastq/
	mv R1_sl.fastq.gz.4tenx.fastq.gz /home/FM/thains/TELLSeq/convertedFastq/PyrrhuraPerlata_S5_L001_R1_001.fastq.gz
	gzip R2_s1.fastq.gz.4tenx.fastq
	mv R2_s1.fastq.gz.4tenx.fastq.gz /home/FM/thains/TELLSeq/convertedFastq/PyrrhuraPerlata_S5_L001_R2_001.fastq.gz

# Supernova Assembly 
The assembly name for species will start with the class it belongs to so b for birds, m for mammals, r for reptiles, a for amphibians, and f for fishes.
then the first three letters for the genus name with the first captialized followed by the first three letters of the species name (again for letter in the species name is capitalized)
this is followed by the number of which this is the version or individual, ask me if you have a question about this.
For example, Pyrrhura perlata would be bPyrPer1. The 1 at the end corresponds to the version of the assembly and it's the first individual. If we did another assembly with a different individual, it would be bPyrPer2.

set the assembly name as asmID

	asmID=bPyrPer1
	
run supernova run
	supernova run --id=${asmID}_supernova --fastq=/home/FM/thains/TELLSeq/convertedFastq --sample=PyrrhuraPerlata--maxreads='all' --accept-extreme-coverage

generate the fasta files for the assembly after supernova has finished with the commands below: pseudohap generates a haploid assembly and pseudohap2 generates a diploid assembly in two fasta files

haploid assembly:
	supernova mkoutput --asmdir=${asmID}_supernova/outs/assembly/ --outprefix=${asmID} --style=pseudohap
	
diploid assembly:
	supernova mkoutput --asmdir=${asmID}_supernova/outs/assembly/ --outprefix=${asmID} --style=pseudohap2


example of bPyrLep1.reads:

	q1=/mnt/nfs/volume1/bucket/Combined_Storage/thains/PyrrhuraLepida_S2_L001_R1_001.fastq.gz
	q2=/mnt/nfs/volume1/bucket/Combined_Storage/thains/PyrrhuraLepida_S2_L001_R2_001.fastq.gz

run scaff_read and follow it up with run trimmomatic or trimgalore to clean reads and this will output all read files from '${asmID}.reads' into the two read files following

	/home/FM/thains/Scaff10X/src/scaff_reads ${asmID}.reads ${asmID}_R1.fastq.gz ${asmID}_R2.fastq.gz

	trimmomatic PE -threads 12 ${asmID}_R1.fastq.gz ${asmID}_R2.fastq.gz -output ${asmID}_trimmed.fastq.gz ILLUMINACLIP:Adapters.fasta:2:30:10:2:keepBothReads LEADING:3 TRAILING 3 MINLEN:36
	mv ${asmID}_trimmed_1P.fastq.gz ${asmID}_R1.trimmed.fastq.gz
	mv ${asmID}_trimmed_2P.fastq.gz ${asmID}_R2.trimmed.fastq.gz
	cat ${asmID}_trimmed_1U.fastq.gz ${asmID}_trimmed_2U.fastq.gz > ${asmID}_unpaired.trimmed.fastq.gz

unzip fasta file generated by supernova mkoutput pseudohap
	gunzip ${asmID}.fasta.gz
	mv ${asmID}.fasta ${asmID}_supernova.fasta

# Removing redundant and duplicate sequences
	bwa index ${asmID}_supernova.fasta
	samtools faidx ${asmID}_supernova.fasta
	bwa mem -t12 ${asmID}_supernova.fasta ${asmID}_R1.trimmed.fastq.gz ${asmID}_R2.trimmed.fastq.gz | samtools view -@ 12 -bSh - > ${asmID}_purgeDups_paired.bam
	bwa mem -t12 ${asmID}_supernova.fasta ${asmID}_unpaired.trimmed.fastq.gz | samtools view -@ 12 -hbS - > ${asmID}_purgeDups_single.bam
	samtools merge -@ 12 ${asmID}_purgeDups.bam ${asmID}_purgeDups_paired.bam ${asmID}_purgeDups_singe.bam
	ngscstat bPyrPer1_purgeDups.bam
	calcuts TX.stat > ${asmID}.cutoffs 2> ${asmID}_calcults.log
	split_fa ${asmID}_supernova.fasta > ${asmID}_supernova.split
	minimap2 -xasm5 -DP ${asmID}_supernova.split ${asmID}_supernovac.split | gzip -c - > ${asmID}_supernova.split.paf.gz
	purge_dups -2 -T ${asmID}.cutoffs -c TX.base.cov ${asmID}_purgeDups1.split.paf.gz > ${asmID}_dups.bed 2> ${asmID}_purgedups.log
	get_seqs -p ${asmID} -e ${asmID}_dups.bed ${asmID}_supernova.fasta
	awk '{if (/>.*/) {print} else { sub(/^N*/, "")sub(/N*$/, ""); print}}' ${asmID}.purged.fa | seqtk seq -l0 -L100 - > ${asmID}.supernova_purged.fasta

## you also should align you linked reas to the assembly using LongRanger
# this requires two steps

	seqtk seq -L5000 <draft.fasta> > <longseq.fasta>

	longranger mkref <longseq>.fasta

	longranger 